src/main/java/org/dependencytrack/persistence/VulnerabilityQueryManager.java

```diff
@@ -142,12 +168,20 @@ public Vulnerability updateVulnerability(Vulnerability transientVulnerability, b
      * @param commitIndex specifies if the search index should be committed (an expensive operation)
      * @return a Vulnerability object
      */
+    @Override
     public Vulnerability synchronizeVulnerability(Vulnerability vulnerability, boolean commitIndex) {
-        Vulnerability result = updateVulnerability(vulnerability, commitIndex);
-        if (result == null) {
-            result = createVulnerability(vulnerability, commitIndex);
+        Vulnerability existingVulnerability  = getExistingVulnerability(vulnerability);
+        if (existingVulnerability != null) {
+            if(hasChanges(existingVulnerability, vulnerability)) {
+                // Update only if changes are detected
+                return updateVulnerability(vulnerability, commitIndex);
+            } else {
+                return null;
+            }
+        } else {
+            // Create new vulnerability if it doesnt exist
+            return  createVulnerability(vulnerability, commitIndex);
```

Comment:
Consider wrapping this in a `callInTransaction` while you're at it. Select and update should share the same transaction.

Reply:
@nscuro thanks for the suggestion, I fix the code.